### DISCUSSION TITLE
Accessing cookies in client component rendering to fetch with authorization

### DISCUSSION THREAD

**Discussion by ellemedit** [vercel/next.js#52785] [Help]
### Summary

In App Router, can we pass request headers cookies into client component fetches default headers?

In users browser, it renders well but if API needs cookie authorization, it is not possible to prerender client component. it enforces fetching in server component, can we change it?

### Example

https://codesandbox.io/p/sandbox/lucid-zhukovsky-xvk382?file=%2Fapp%2Fpage.tsx%3A8%2C15

**Comment by devjiwonchoi** [comment]
For client-side cookies, you can implement third-party modules like [js-cookie](https://www.npmjs.com/package/js-cookie) and will work like a charm.

**Reply by ellemedit** [reply]
I need to read cookie in prerender phase in client component. That doesn't solve this problem.

**Comment by MarioMS90** [comment]
Is there any update on this issue? Apparently, it is impossible to read any cookies if we make a fetch during the client prerender phase.
